src/scripts/clipperUI/components/sectionPicker.tsx

```diff
@@ -210,16 +210,12 @@ export class SectionPickerClass extends ComponentBase<SectionPickerState, Sectio
 			this.setDataSource();
 		}
 
-		return new Promise<OneNoteApi.ResponsePackage<OneNoteApi.Notebook[]>>((resolve, reject) => {
-			let headers: { [key: string]: string } = {};
-			headers[Constants.HeaderValues.appIdKey] = Settings.getSetting("App_Id");
-			headers[Constants.HeaderValues.userSessionIdKey] = sessionId;
-
-			SectionPickerClass.dataSource.getNotebooks(headers).then((responsePackage: OneNoteApi.ResponsePackage<OneNoteApi.Notebook[]>) => {
-				resolve(responsePackage);
-			}, (failure: OneNoteApi.RequestError) => {
-				reject(failure);
-			});
+		let headers: { [key: string]: string } = {};
+		headers[Constants.HeaderValues.appIdKey] = Settings.getSetting("App_Id");
+		headers[Constants.HeaderValues.userSessionIdKey] = sessionId;
+
+		return SectionPickerClass.dataSource.getNotebooks(headers).catch((failure: OneNoteApi.RequestError) => {
```

Comment:
The `catch` is unnecessary here. Returning the promise will propagate the error already.